src/main/scala/flow.scala

```diff
@@ -101,16 +102,24 @@ sealed abstract class Comprehension[M[_]]{ // FIXME: what happens if calling a m
   //def apply[T](scope: M[FlowContext] => T): M[T]
   //def apply[T](scope: => T): M[T]
 }
+final class MonadContext[M[_]]{
+  @compileTimeOnly("The MonadContext type only makes sense in a flow comprehension scope and is supposed to be removed by the macro.")
+  def apply(transform: M[FlowContext] => M[FlowContext]) = ???
+}
 class sequence[M[_]] extends Comprehension[M]{
-  def apply[T](comprehension: M[FlowContext] => T): M[T] = macro FlowMacros.sequence[M[_],T]
+  def apply[T](comprehension: MonadContext[M] => T): M[T] = macro FlowMacros.sequence[M[_],T]
   //def apply[T](scope: => T): M[T] = ???
 }
```

Comment:
Took me a minute to follow this change, but yeah, it makes sense given the desired syntax change.